public/modules/custom/infofinland_user_cancel/infofinland_user_cancel.module

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @file
+ * Helper functions related to deleting user accounts.
+ */
+
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_user_cancel().
+ */
+function infofinland_user_cancel_user_cancel($edit, UserInterface $account, $method) {
+  switch ($method) {
+    case 'user_cancel_reassign':
+      // Anonymize all of the nodes for this old account.
+      _infofinland_user_cancel_reassign_nodes_to_anonymous($account);
+      break;
+  }
+}
+
+/**
+ * Reassigns all node revisions by $account to by owned by anonymous (uid 0).
+ *
+ * Replicating functionality of node module, because node code is intended to
+ * be run from a form or otherwise batched.
+ *
+ * @param \Drupal\user\UserInterface $account
+ *   The user.
+ */
+function _infofinland_user_cancel_reassign_nodes_to_anonymous(UserInterface $account): void {
```

Comment:
Have you benchmarked this? I'm wondering if this should be queued here and processed in a queue worker instead

Reply:
Even with one of the most problematic users that had about 5000 - 7000 revisions this completed pretty quickly because it's a simple database query. Might be worth adding a batch processor if this causes memory issues later, but not sure if it's required now?

Reply:
Should be fast enough for now then. I kinda assumed it would take 30-60sec+